src/main/java/g0101_0200/s0160_intersection_of_two_linked_lists/Solution.java

```diff
@@ -4,15 +4,12 @@
 
 public class Solution {
     public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
-        ListNode pA = headA;
-        ListNode pB = headB;
-        while (pA != null || pB != null) {
-            if (pA != null && pB != null && pA.toString().equals(pB.toString())) {
-                return pA;
-            }
-            pA = pA == null ? headB : pA.next;
-            pB = pB == null ? headA : pB.next;
+        ListNode node1 = headA;
+        ListNode node2 = headB;
+        while (!node1.toString().equals(node2.toString())) {
```

Comment:
Why toString?

Reply:
The above implementation, that you wrote results in nullpoinerexception in the test case, as node1 == node2 evaluates to false

Reply:
I tested it in leetcode. There no exceptions.

![image](https://user-images.githubusercontent.com/348496/142396054-c6d3218b-2e3f-4d1c-ac57-f7819af9614d.png)